finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ServerBridge.scala

```diff
@@ -15,6 +17,17 @@ private[netty4] class ServerBridge[In, Out](
 
   def initChannel(ch: Channel): Unit = {
     val transport: Transport[In, Out] = transportFac(ch)
-    serveTransport(transport)
```

Comment:
I'm not so convinced :). Why should it, if we explicitly configured channel do not read any data  unless we explicitly tell it to do so. And `SslHandler` has no idea, if we want to serve the channel or not.
But anyway, the workaround you suggested seems to work: https://github.com/koshelev/finagle/commit/282ca2c5f281fabcdaa22495a07d1cfd850628ad